app/src/main/java/ch/epfl/sdp/social/ChatRepository.java

```diff
@@ -174,34 +174,54 @@ protected void onPostExecute(List<Message> ls)
 
 
 
-    public static void insertMessageFromRemote(Timestamp tm, String content) {
+    public static void insertMessageFromRemote(Timestamp tm, String content, int chat_id) {
         new AsyncTask<Void, Void, Void>() {
             @Override
             protected Void doInBackground(Void... voids) {
-                //try {
-                    singleton.chatDB.daoAccess().sendMessage(new Message(tm.toDate(), content));
-                //}catch (SQLiteConstraintException e)
-                //{ }
+                try {
+                    singleton.chatDB.daoAccess().sendMessage(new Message(tm.toDate(), content, chat_id));
+                }catch (SQLiteConstraintException e)
+                { }
                 return null;
             }
         }.execute();
 
     }
 
-    public static void getChat(String current, String other)
+    public static Chat getChat(String current, String other)
     {
-        new AsyncTask<Void, Void, List<Chat>>() {
-
-            @Override
-            protected List<Chat> doInBackground(Void... voids) {
-                return singleton.chatDB.daoAccess().getChatFromCurrentToOther(current, other);
-            }
+        Chat output = null;
+        try {
+            output = new AsyncTask<Void, Void, List<Chat>>() {
+
+                @Override
+                protected List<Chat> doInBackground(Void... voids) {
+                    return singleton.chatDB.daoAccess().getChatFromCurrentToOther(current, other);
+                }
+
+                /*@Override
+                protected void onPostExecute(List<Chat> exists) {
+                    ((WaitOnChatRetrieval)(singleton.contextActivity)).chatFetched(exists);
+                }*/
+            }.execute().get().get(0);
+        } catch (ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+        }
+        return output;
+    }
 
-            @Override
-            protected void onPostExecute(List<Chat> exists)
-            {
-                ((WaitOnChatRetrieval)(singleton.contextActivity)).chatFetched(exists);
-            }
-        }.execute();
+    public Chat getChatDirectly(String current, String other) {
```

Comment:
getChatDirectly? As in getChat was getting the chat _indirectly_? To me this method seems like it is almost doing the same thing as getChat so either consider removing getChatDirectly or modifying getChat to accommodate the functionality you want from getChatDirectly.

Reply:
This is a leftover from another solution, I will remove it.